src/stream/src/common/table/state_table.rs

```diff
@@ -514,7 +525,9 @@ impl<S: StateStore> StateTable<S> {
         std::mem::replace(&mut self.vnodes, new_vnodes)
     }
 }
-
+fn is_subset(vec1: Vec<usize>, vec2: Vec<usize>) -> bool {
+    HashSet::<usize>::from_iter(vec1).is_subset(&vec2.into_iter().collect())
```

Comment:
I think we need a stricter check here instead of just `is_subset` because `dist_key_in_pk_indicies` needs to be a continous subset of `pk_indicies`. We can actually do this check efficiently:
```
fn is_continous_subset<T: Eq>(
    mut sub_iter: impl Iterator<Item = T>,
    mut iter: impl Iterator<Item = T>,
) -> bool {
    if let Some(first) = sub_iter.next() {
        if iter.find(|item| item == &first).is_none() {
            return false;
        }

        for sub_item in sub_iter {
            if let Some(item) = iter.next() && item == sub_item {
                continue;
            }
            return false;
        }
    }
    true
}
```

Reply:
With this check, we can simplify L1027-L1030.

Reply:
Good idea!